src/nanoarrow/nanoarrow_types.h

```diff
@@ -656,6 +672,19 @@ struct ArrowArrayPrivateData {
   int8_t union_type_id_is_child_index;
 };
 
+/// \brief A representation of an interval.
+struct ArrowInterval {
+  /// \brief The type of interval being used
+  enum ArrowIntervalUnit unit;
+
+  /// \brief Pointer containing the underlying data for the interval
+  union {
+    int32_t year_month;
+    struct ArrowIntervalDayTime day_time;
+    struct ArrowIntervalMonthDayNano month_day_nano;
+  } data;
+};
```

Comment:
`time` is supposed to represent milliseconds in that struct right? Not a strong preference but I think using the structs is a little easier as an end user. With the current design I was thinking an anonymous union would be the best, though that would require C11 instead of just C99

Reply:
Yea makes sense. Actually looking in CI I don't think Cython supports using a union within a struct, so this is probably the only way

Reply:
Even with flattening to one struct I think we will still need to keep structs like `ArrowIntervalDayTime` and `ArrowIntervalMonthDayNano` to unpack the data buffer - is that in line with your expectation or am I misinterpreting?

Reply:
Nice catch - simplified that